### PR TITLE
Kilian/adaptive curriculum

### LINKED ISSUE #24: adaptive curriculum
sample data depending on next task
inside dataloader

### PR DESCRIPTION
**Pull request by KilianFt**

### CODE REVIEW

**Comment by KilianFt** [comment]
can merge once replay buffer is merged

File: adaptive_hci/datasets.py [hunk 1]
```diff
@@ -29,6 +30,33 @@
...
+        modes.append(main_action)
+
+    return modes
+
+
+def find_closest_episode(episodes, target_row):
+    ep_modes = get_episode_modes(episodes)
+
+    distances = np.linalg.norm(ep_modes - target_row, axis=1)
+    min_distance_indices = np.where(distances == distances.min())[0]
```
Comment (manuel-delverme) [inline adaptive_hci/datasets.py:47]: `    distances = np.argmin(np.linalg.norm(ep_modes - target_row, axis=1))`

File: adaptive_hci/datasets.py [hunk 2]
```diff
@@ -29,6 +30,33 @@
 ]
 
 
+def get_episode_modes(episodes, n_samples_considered: int = 20):
+    modes = []
+    for ep in episodes:
```
Comment (manuel-delverme) [inline adaptive_hci/datasets.py:35]: when we have a leaderboard, check if mixing the episodes and taking the most common over that is better

**Review by manuel-delverme — APPROVED** [review]

File: adaptive_hci/datasets.py (hunk 1 above)
Comment (KilianFt) [reply adaptive_hci/datasets.py]: argmin into random.choice crashed, I think only taking argmin is fine as well, shouldn't change much

File: adaptive_hci/datasets.py (hunk 2 above)
Comment (KilianFt) [reply adaptive_hci/datasets.py]: yes that should be the next step to do some sweep and find out

**Pull request merged**